apis/placement/v1alpha1/stagerollout_types.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",categories={fleet,fleet-placement}
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StageRollout defines a group of override policies about how to override the selected cluster scope resources
+// to target clusters.
+type StageRollout struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of StageRollout.
+	// +required
+	Spec StageRolloutSpec `json:"spec"`
+
+	// The observed status of StageRollout.
+	// +required
+	Status StageRolloutStatus `json:"status"`
+}
+
+// StageRolloutSpec defines the desired the rollout sequence stage by stage.
+type StageRolloutSpec struct {
+	// Stage rollout configurations for each rollout stage.
+	// +required
+	StageRollout []StageRolloutConfig `json:"stages"`
+
+	// The wait time between each stage is completed.
+	// +optional
+	WaitTimeBetweenStage *metav1.Duration `json:"waitTimeBetweenStage,omitempty"`
+
+	// TODO: Add alerting configuration.
+
+	// TODO: Add health-check configuration.
```

Comment:
Hi Ryan! For the P0 we would also need at least a manual gate between stages.

Reply:
just to be clear, it's a "hook" to inject "go ahead" signal to our rollout between stages.